src/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,323 @@
+import React, {forwardRef} from 'react'
+import {
+  Text,
+  TextInput as Input,
+  StyleProp,
+  ViewStyle,
+  TextStyle,
+  TextProps,
+  ColorValue,
+  ReturnKeyTypeOptions,
+  ImageSourcePropType,
+  ImageResizeMode,
+  ImageStyle,
+  KeyboardTypeOptions,
+} from 'react-native'
+import {metrics} from '../helpers/metrics'
+import {Icon} from '../Icon/Icon'
+import styled from 'styled-components/native'
+import {colors} from '../helpers/colors'
+
+export interface TextInputProps {
+  /** Style for container */
+  containerStyle?: StyleProp<ViewStyle>
+
+  /** If false, text is not editable. The default value is true. */
+  editable?: boolean
+
+  /** Styling for Input Component Container */
+  inputContainerStyle?: StyleProp<ViewStyle>
+
+  /** Style for Input Component */
+  inputStyle?: StyleProp<TextStyle>
+
+  /** Value of the text input */
+  value?: string
+
+  /** Add a label on top of the input */
+  label?: string
+
+  /** Add star beside the label */
+  isRequire?: boolean
+
+  /** Styling for the label. You can only use this if label is a string */
+  labelStyle?: StyleProp<TextStyle>
+
+  /** Props to be passed to the React Native Text component used to display the label or React Component used instead of simple string in label prop */
+  labelProps?: TextProps
+
+  /** Displays a left or right icon. (can be used along with iconPosition as well) */
+  icon?: ImageSourcePropType
+
+  /** Displays Icon size need to be used along with `icon` prop */
+  iconSize?: number
+
+  /** Displays Icon color need to be used along with `icon` prop */
+  iconColor?: string
+
+  /** Determines how to resize the image when the frame doesn't match the raw */
+  resizeMode?: ImageResizeMode
```

Comment:
Change this to `iconResizeMode`

Reply:
ImageResizeMode type is defined by React Nativ and I think we should not change its name